src/app/api/crud.py

```diff
@@ -40,3 +45,31 @@ async def put(entry_id: int, payload: BaseModel, table: Table):
 async def delete(entry_id: int, table: Table):
     query = table.delete().where(entry_id == table.c.id)
     return await database.execute(query=query)
+
+
+class DeviceCRUD:
+
+    @classmethod
+    async def heartbeat(cls, device: DeviceOut) -> HeartbeatOut:
+        device.last_ping = datetime.utcnow()
+        await put(device.id, device, devices)
+        return device
+
+    @classmethod
+    async def user_owns_device(cls, device_id: int, owner_id: int) -> bool:
+        query = devices.select().where(and_(devices.c.id == device_id, devices.c.owner_id == owner_id))
+        return bool(await database.fetch_one(query=query))
+
+    @classmethod
+    async def update_location(cls, payload: UpdatedLocation, device_id: int, user_id: int):
+        user_owns_device = await cls.user_owns_device(device_id, user_id)
+        if not user_owns_device:
+            raise HTTPException(
+                status_code=400,
+                detail="You don't own this device."
+            )
+        device = (await get(device_id, devices))
+        device.update(payload.dict())
+        device = DeviceOut(**device)
+        await put(device.id, device, devices)
+        return device
```

Comment:
Before moving on a proper refactoring, I would have 2 suggestions :
1. Specific tables (i.e. "devices" here) might be better off being imported in a single file to make later refactoring easier (for devices, I think it's only imported /passed in `routes/devices.py`). Maybe the `device_table` could be passed here?
2. Since CRUD is supposed to be basic table operations, perhaps we could move these three methods into `routing.py` directly? (since we will have to split either CRUD or routing, let's put everything in routing for now for easier refactoring later. Routing already has table-specific functions)

What do you think?

Reply:
2. I agree. I might keep CRUD as it is with basic operations and move methods to routing (to refactor later) because it is already specifc.

Reply:
I moved the code from CRUD to routing. 
It implied a bit refactor for fetch_one and fetch_all to add the possibility to have multiple_conditions.